### PR TITLE
Allow all members of private orgs to see public repos

### LINKED ISSUE #10144: Public repositories within private org are not visible to members
- Gitea version (or commit ref): 1.10.3 / 1.12.0+dev-251-gd4096ab6a
- Git version: Not relevant
- Operating system: Not relevant
- Database:
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist: Not relevant

## Description

When repository, within an organization which has visibility set to Private, is set as public and user is part of a team that has no access specified to this repository, repository can be accessed via direct URI but can not be viewed on any list within UI.

I would expect that setting repository as public within Private organization would allow members of all teams to view it, regardless of team's permissions and team's repositories (for example I might want to make repository visible but read-only to all members of organization, while certain teams can write to different repositories).

This does not apply when organization is set as Limited. In such a case, repository is properly visible (to both members of organization as well as outside users).

Owners of org can of course see all repositories just fine.

## Screenshots

chrome_2020-02-05_01-32-09
chrome_2020-02-05_01-32-25
chrome_2020-02-05_01-32-38
chrome_2020-02-05_01-32-46
chrome_2020-02-05_01-32-52
firefox_2020-02-05_01-31-20
firefox_2020-02-05_01-31-33
firefox_2020-02-05_01-31-57

### PR DESCRIPTION
**Pull request by zeripath**
Adjust accesible repo condition to allow members of private orgs to see their public repos.

Fix #10144

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: models/repo_list.go [hunk 1]
```diff
@@ -401,21 +401,26 @@ func accessibleRepositoryCondition(user *User) builder.Cond {
...
-			Where(builder.Eq{"`team_user`.uid": user.ID}).
-			Join("INNER", "team_user", "`team_user`.team_id = `team_repo`.team_id")))
+			// 3. Repositories that we directly own
+			builder.Eq{"`repository`.owner_id": user.ID},
+			// 4. Be able to see all repositories that we are in a team
+			builder.In("`repository`.id", builder.Select("`team_repo`.repo_id").
+				From("team_repo").
+				Where(builder.Eq{"`team_user`.uid": user.ID}).
+				Join("INNER", "team_user", "`team_user`.team_id = `team_repo`.team_id")),
+			// 5. Be able to see all public repos in public repos of organizations that we are an org_user of
```
Comment (lunny) [inline models/repo_list.go]: private?

Comment (lafriks) [inline models/repo_list.go]:
```suggestion
			// 5. Be able to see all public repos in private organizations that we are an org_user of
```

**Review by lafriks — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Comment by lunny** [comment]
Please send back port to v1.11

**Comment by zeripath** [comment]
Damn that was a hard backport.

**Pull request merged**